backend/btrixcloud/subs.py

```
@@ -378,14 +382,77 @@ async def get_billing_portal_url(
+                        return AddonMinutesPricing(**json)
+            # pylint: disable=broad-exception-caught
+            except Exception as exc:
+                print("Error fetching execution minutes price", exc)
+
+    async def get_checkout_url(
+        self,
+        org: Organization,
+        headers: dict[str, str],
+        minutes: int | None,
+    ):
+        """Create checkout url for additional minutes"""
+        if not org.subscription:
+            raise HTTPException(
+                status_code=404, detail="Organization has no subscription"
+            )
+        subscription_id = org.subscription.subId
+        return_url = f"{get_origin(headers)}/orgs/{org.slug}/settings/billing"
 
-# pylint: disable=invalid-name,too-many-arguments
+        if external_subs_app_api_url:
+            try:
+                req = CheckoutAddonMinutesRequest(
+                    orgId=str(org.id),
+                    subId=subscription_id,
+                    minutes=minutes,
+                    return_url=return_url,
+                )
+                async with aiohttp.ClientSession() as session:
+                    async with session.request(
+                        "POST",
+                        f"{external_subs_app_api_url}/checkout/additionalMinutes",
+                        headers={
+                            "Authorization": "bearer " + external_subs_app_api_key,
+                            "Content-Type": "application/json",
+                        },
+                        json=req.model_dump(),
+                        raise_for_status=True,
+                    ) as resp:
+                        json = await resp.json()
+                        print(f"get_checkout_url got response: {json}")
+                        return CheckoutAddonMinutesResponse(**json)
+            # pylint: disable=broad-exception-caught
+            except Exception as exc:
+                print("Error fetching checkout url", exc)
```

I wonder if we should (re)raise the exception here so that the failure case is documented in the API docs rather than having the potential to return `None`